sdk/tests/conformance/misc/webgl-specific.html

```diff
@@ -78,23 +78,59 @@
 
 debug("");
 debug("Verify that front/back settings should be the same for stenclMask and stencilFunc");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMask(255)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMaskSeparate(gl.FRONT, 1)");
-wtu.shouldGenerateGLError(gl, gl.INVALID_OPERATION, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMaskSeparate(gl.BACK, 1)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFunc(gl.ALWAYS, 0, 255)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFuncSeparate(gl.BACK, gl.ALWAYS, 1, 255)");
-wtu.shouldGenerateGLError(gl, gl.INVALID_OPERATION, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFuncSeparate(gl.FRONT, gl.ALWAYS, 1, 255)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFuncSeparate(gl.BACK, gl.ALWAYS, 1, 1)");
-wtu.shouldGenerateGLError(gl, gl.INVALID_OPERATION, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFuncSeparate(gl.FRONT, gl.ALWAYS, 1, 1)");
-wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.TRIANGLES, 0, 0)");
+const fb = gl.createFramebuffer();
+gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+const colorRB = gl.createRenderbuffer();
+gl.bindRenderbuffer(gl.RENDERBUFFER, colorRB);
+gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA4, 1, 1);
+gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, colorRB);
+
+function testForStencilBits(errIfMismatch) {
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.POINTS, 0, 0)");
+
+
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMaskSeparate(gl.FRONT, 1)");
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMaskSeparate(gl.BACK, 257)");
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.POINTS, 0, 0)");
+
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMaskSeparate(gl.BACK, 258)");
+    wtu.shouldGenerateGLError(gl, errIfMismatch, "gl.drawArrays(gl.POINTS, 0, 0)");
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilMask(1023)");
+
+
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFuncSeparate(gl.FRONT, gl.ALWAYS, 1, 1023)");
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.stencilFuncSeparate(gl.BACK, gl.ALWAYS, 257, 1023)");
+    wtu.shouldGenerateGLError(gl, gl.NO_ERROR, "gl.drawArrays(gl.POINTS, 0, 0)");
```

Comment:
@jdashg , I tried running your new test with the ANGLE running in webglCompatibility mode and received several failures.
This fourth call to `drawArrays` fails with INVALID_OPERATION in the following ANGLE code in `ValidateDrawBase`

````C++
// Note: these separate values are not supported in WebGL, due to D3D's limitations. See
// Section 6.10 of the WebGL 1.0 spec.
Framebuffer *framebuffer = state.getDrawFramebuffer();
if (context->getLimitations().noSeparateStencilRefsAndMasks || extensions.webglCompatibility)
{
    const FramebufferAttachment *dsAttachment =
        framebuffer->getStencilOrDepthStencilAttachment();
    GLuint stencilBits                = dsAttachment ? dsAttachment->getStencilSize() : 0;
    GLuint minimumRequiredStencilMask = (1 << stencilBits) - 1;
    const DepthStencilState &depthStencilState = state.getDepthStencilState();

    bool differentRefs = state.getStencilRef() != state.getStencilBackRef();
    bool differentWritemasks =
            (depthStencilState.stencilWritemask & minimumRequiredStencilMask) !=
            (depthStencilState.stencilBackWritemask & minimumRequiredStencilMask);
    bool differentMasks = (depthStencilState.stencilMask & minimumRequiredStencilMask) !=
                              (depthStencilState.stencilBackMask & minimumRequiredStencilMask);

    if (differentRefs || differentWritemasks || differentMasks)
    {
        if (!extensions.webglCompatibility)
        {
            ERR() << "This ANGLE implementation does not support separate front/back stencil "
                     "writemasks, reference values, or stencil mask values.";
        }
        ANGLE_VALIDATION_ERR(context, InvalidOperation(), StencilReferenceMaskOrMismatch);
        return false;
    }
}
````
Though `differentWritemasks` and `differentMasks` are false, `differentRefs` is true. The stencil ref is 1 and the stencil back ref is 257. Looks like the minimum required bits only applies to the masks, not the reference value.

Reply:
It seems like a bug in ANGLE that the minimumRequiredStencilMask isn't applied to the reference values. @RafaelCintron, do you agree? Is applying this mask compatible with Direct3D's semantics?

Reply:
@kenrussell , D3D11 does not support creating textures with more than 8bits of stencil data.  Hence, the read and write masks are only limited to 8 bits.  See the [DEPTH_STENCIL_DESC](https://msdn.microsoft.com/en-us/library/windows/desktop/ff476110(v=vs.85).aspx) documentation.  The reference value is an unsigned integer. See the [OMSetDepthStencilState](https://msdn.microsoft.com/en-us/library/windows/desktop/ff476463(v=vs.85).aspx) for documentation.

In [`StateManager11::syncDepthStencilState`](https://github.com/google/angle/blob/master/src/libANGLE/renderer/d3d/d3d11/StateManager11.cpp#L1084), ANGLE clamps the reference value before calling D3D11 with a comment stating this is because of the GL spec.  

I suppose It would be nice to inform the web developer when bits of their reference value is not taking effect but I do not feel strongly about that.